src/lib.rs

```diff
@@ -413,6 +416,126 @@ impl Contract {
+        }
+        addon.validate();
+        self.available_addons.insert(&addon.id.clone(), &addon);
+    }
+
+    // ONLY FOR TESTING
+    pub fn delete_addon(&mut self, id: CommunityAddOnId) {
+        // Also delete from communities
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can delete add-ons");
+        }
+        let addon = self
+            .get_addon(id.clone())
+            .expect(&format!("Add-on with id `{}` does not exist", id))
+            .clone();
+
+        // let mut new_communities: UnorderedMap<CommunityHandle, Community> =
+        //     UnorderedMap::new(StorageKey::Communities);
+
+        // for (community_handle, mut community) in self.communities.iter() {
+        //     // Try to remove add on from community
+        //     community.remove_addon(addon.id.to_owned());
+        //     new_communities.insert(&community_handle, &community);
+        // }
+        // self.communities = new_communities;
+        self.available_addons.remove(&addon.id);
+    }
+
+    pub fn edit_addon(&mut self, input: CommunityAddOn) {
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can edit add-ons");
+        }
+        self.available_addons.insert(&input.id.clone(), &input);
+    }
+
+    pub fn get_community_addons(&self, handle: CommunityHandle) -> Vec<CommunityAddOn> {
+        let community = self
+            .get_community(handle.to_owned())
+            .expect(format!("Community with handle `{}` does not exist", handle).as_str());
+        community
+            .addon_list
+            .iter()
+            .map(|addon_config| {
+                self.get_addon(addon_config.addon_id.clone()).expect("add-on missing")
+            })
+            .collect()
+    }
+
+    pub fn add_community_addon(
+        &mut self,
+        community_handle: CommunityHandle,
+        addon_config: CommunityAddOnConfig,
+    ) {
+        let mut community = self.get_community(community_handle.to_owned()).expect(
+            format!("Community with handle `{}` does not exist", community_handle).as_str(),
+        );
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+            && !community.admins.contains(&env::predecessor_account_id())
+        {
+            panic!("Only moderators and community admins can add/edit add-ons to a community");
+        }
+        if self.get_addon(addon_config.addon_id.to_owned()).is_none() {
+            panic!("No add-on exists with this id");
+        }
+
+        community.add_addon(addon_config);
+        self.communities.insert(&community_handle, &community);
+    }
+
+    pub fn edit_community_addon(
```

Comment:
"update_community_addon" may be a better name

Reply:
Will removing and then creating modify the addon's id? And so references elsewhere will break once edited?

Reply:
You could in theory update the `addon_id` in CommunityAddOnConfig. It would point to a different add on. However not the `config_id` (because remove_ would panic) which wouldn't break the reference made by peters comment.